packages/labs/src/components/tag-input/tagInput.tsx

```diff
@@ -264,12 +271,13 @@ export class TagInput extends AbstractComponent<ITagInputProps, ITagInputState>
     private handleInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
         const { selectionEnd, value } = event.currentTarget;
         if (event.which === Keys.ENTER && value.length > 0) {
+            const { onAdd, onChange, values } = this.props;
             // enter key on non-empty string invokes onAdd
             const newValues = this.getValues(value);
-            let shouldClearInput = Utils.safeInvoke(this.props.onAdd, newValues);
+            let shouldClearInput = Utils.safeInvoke(onAdd, newValues);
             // avoid a potentially expensive computation if this prop is omitted
-            if (Utils.isFunction(this.props.onChange)) {
-                shouldClearInput = shouldClearInput || this.props.onChange([...this.props.values, ...newValues]);
+            if (Utils.isFunction(onChange)) {
+                shouldClearInput = shouldClearInput || onChange([...values, ...newValues]);
```

Comment:
@adidahiya suggestions? this _works_, and there's gonna be assignment regardless.